packages/contracts/contracts/RollupChain.sol

```diff
@@ -0,0 +1,222 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+/* Internal Imports */
+import {DataTypes as dt} from "./DataTypes.sol";
+import {SparseMerkleTreeLib} from "./SparseMerkleTreeLib.sol";
+import {TransitionEvaluator} from "./TransitionEvaluator.sol";
+
+contract RollupChain {
+    /* Fields */
+    // The Evaluator for our STF
+    TransitionEvaluator transitionEvaluator;
+    dt.Block[] public blocks;
+    bytes32 public ZERO_BYTES32 = 0x0000000000000000000000000000000000000000000000000000000000000000;
+    bytes32[3] private FAILED_TX_OUTPUT = [ZERO_BYTES32, ZERO_BYTES32, ZERO_BYTES32];
+    bool public halted = false;
+    // Tx types
+    uint NEW_ACCOUNT_TRANSFER_TYPE = 0;
+    uint STORED_ACCOUNT_TRANSFER_TYPE = 1;
+    uint SWAP_TYPE = 2;
```

Comment:
Turns out (hilariously) that `uint` is actually cheaper than `uint32` because all data is stored in 32 byte chunks. So smaller units are more expensive because they require transforming a 32 byte word.

Reply:
Hahaha, no kidding. 🌈 ➡️ ⭐️ The more you know.